### ISSUE TITLE
Bad message when inheriting an implementation with dependencies

### ISSUE THREAD

**Issue by buzden** [idris-lang/Idris2#1532]
Consider two interfaces, one being a subinterface of another.

```idris
interface X a where
      
interface X a => Y a where
```

Consider a named implementation of the parent interface having non-empty requirements:

```idris
[SomeX] X a => X (List a) where
```

Create an implementation of the subinterface with explicitly set implementation of the superinterface using `using` syntax:

```idris
[SomeY] Y (List a) using SomeX where
```

You get

```
Error: While processing right hand side of SomeY. Can't find an implementation for X (List a).

...:7:1--7:37
 3 | interface X a => Y a where
 4 |
 5 | [SomeX] X a => X (List a) where
 6 |
 7 | [SomeY] Y (List a) using SomeX where
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

which makes me think: "hey, I've already said which implementation of `X (List a)` I want in the `using` section, what else do you want?"

But after repeating all the requirements of the implementation mentioned in the `using` section, all typechecks:

```idris
[SomeY] X a => Y (List a) using SomeX where
```

Well, it is discussible whether it is good or not that I should repeat all the requirements in the implementation of the subinterface, but still, the error message looks strange.

**Comment by gallais** [comment]
> I should repeat all the requirements in the implementation of the subinterface

I think you should. Otherwise it may be the case that additional requirements are
sneaking past you when upstream changes the definition of the interface and suddenly
you'll have very confusing error messages at use sites.


Now wrt to the error message itself, the question is: should the failure of interface
resolution be explained by detailing all of the attempts that were made and the reason
why none of them were successful? It's probably too noisy in general but we may want
to explicitly complain about useless `using` arguments.

**Comment by buzden** [comment]
> It's probably too noisy in general but we may want
> to explicitly complain about useless `using` arguments.

Yes, I was thinking mostly about this case. Something like "cannot use the implementation mentioned in the `using` clause because of unsatisfied requirements `x` and `y`".
